apps/minifront/src/components/shared/error-boundary.tsx

```diff
@@ -1,17 +1,25 @@
-import { useRouteError } from 'react-router-dom';
+import { isRouteErrorResponse, useRouteError } from 'react-router-dom';
 import { PraxNotConnectedError } from '@penumbra-zone/client';
 import { ExtensionNotConnected } from '../extension-not-connected';
+import { NotFound } from '../not-found';
+import { ExtensionUnavailable } from '../extension-unavailable';
+import { Code, ConnectError } from '@connectrpc/connect';
+import { SplashPage } from '@penumbra-zone/ui';
 
 export const ErrorBoundary = () => {
   const error = useRouteError();
 
+  if (error instanceof ConnectError && error.code === Code.Unavailable) {
+    return <ExtensionUnavailable />;
+  }
   if (error instanceof PraxNotConnectedError) return <ExtensionNotConnected />;
+  if (isRouteErrorResponse(error) && error.status === 404) return <NotFound />;
 
-  console.error(error);
+  console.error('ErrorBoundary caught error:', error);
 
   return (
-    <div className='text-red'>
-      <h1 className='text-xl'>{String(error)}</h1>
-    </div>
+    <SplashPage title='Uh-oh!' description='Looks like there was an error.'>
+      {String(error)}
+    </SplashPage>
```

Comment:
maybe this is inconsequential. but i (and a lot of users) find 'uh-oh' type error pages weirdly infuriating, and i would personally prefer to have a drier header phrase